src/sync.cpp

```diff
@@ -3263,23 +3050,26 @@ bool JSONSyncConfigIOContext::deserialize(JSONSyncConfigMap& configs,
     // Deserialize the configs.
     while (reader.enterobject())
     {
-        JSONSyncConfig config;
+        SyncConfig config;
 
-        if (!deserialize(config, reader))
+        if (deserialize(config, reader))
         {
-            return false;
-        }
+            // So move is well-defined.
+            const auto bid = config.getBackupId();
 
-        // So move is well-defined.
-        const auto bid = config.backupId;
-
-        configs.emplace(bid, std::move(config));
+            configs.emplace(bid, std::move(config));
+        }
+        else
+        {
+            LOG_err << "Failed to deserialize a sync config";
```

Comment:
shouldn't we abort here (return false;) ?

Reply:
The file can have several sync configs in it, if one fails to load I don't think we should abandon the others, we may still be able to deserialize one or more of those?